data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/converter/StreamRecordConverter.java

```diff
@@ -116,4 +121,19 @@ private Map<String, Object> convertKeys(Map<String, AttributeValue> keys) {
         return result;
 
     }
+
+    private long calculateTieBreakingVersionFromTimestamp(final Instant eventTimeInSeconds) {
+        if (currentSecond == null) {
+            currentSecond = eventTimeInSeconds;
+        } else if (currentSecond.isAfter(eventTimeInSeconds)) {
+            return eventTimeInSeconds.toEpochMilli() * 1000;
+        } else if (currentSecond.isBefore(eventTimeInSeconds)) {
+            recordsSeenThisSecond = 0;
+            currentSecond = eventTimeInSeconds;
+        } else {
+            recordsSeenThisSecond++;
+        }
+
+        return eventTimeInSeconds.toEpochMilli() * 1000 + recordsSeenThisSecond;
```

Comment:
What happens when there are more than 1000 records in a second?

Reply:
It would be an issue, but we would have to get 1000 records in same shard, which is impossible at this time